### PR TITLE
Key setup

### PR DESCRIPTION
**Pull request by maelle**
Cf #691 

Notes

* The change in DESCRIPTION is just my adding rOpenSci as fnd via `rodev::add_ro_fnd()` which adds many new lines. Oh and I use `roxygen2` dev version.

* I might have deleted too much from getkey docs?

* I wonder whether the functions are a bit misleading, giving the wrong impression than when requesting a key, you'll get it right away?

* I wasn't too sure how to test these functions.

* use_iucn should maybe live in `rredlist` instead.

### CODE REVIEW

**Comment by maelle** [comment]
I also wanted to add that if/when `usethis` exports its styling functions, the messages could look nicer with not too much effort.

File: R/getkey.R [hunk 1]
```diff
@@ -35,32 +35,6 @@
...
-#' session by running \code{options(EOL_KEY = "<your eol api key>")} in
-#' the console.
-#'
-#' @section IUCN:
-#' IUCN requires an API key. See \code{\link[rredlist]{rredlist-package}} 
-#' for help on authentiating with IUCN Redlist, and 
-#' http://apiv3.iucnredlist.org/api/v3/token for requesting a key
-#' 
-#' @section NCBI Entrez:
-#' From NCBI's docs: "E-utils users are allowed 3 requests/second without an
```
Comment (sckott) [inline R/getkey.R]: @maelle i think we do want to retain this bit about 3 reqeusts per second without a key, and that using a key increases rate limit

Comment (sckott) [reply R/getkey.R]: the other lines deleted are okay

**Comment by sckott** [comment]
* the changes in description are fine
* "I wonder whether the functions are a bit misleading, giving the wrong impression than when requesting a key, you'll get it right away?" -> Can you add some text to the key_helpers man file reinforcing this?
* "use_iucn should maybe live in rredlist instead.", yeah, that's a good idea, then I guess we can just point users to `rredlist::use_iucn()` instead of `use_iucn()`

File: R/key_helpers.R [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+#' but you should get higher rate limit with a key, so do get one.
+#'  
+#' @export
+#' @rdname key_helpers
+use_entrez <- function(){
+  if(interactive()){
+    utils::browseURL("https://www.ncbi.nlm.nih.gov/account/")
+  }
+  
+  message("Create your key from your (brand-new) account's. \nAfter generating your key set it as ENTREZ_KEY in .Renviron.\n ENTREZ_KEY='youractualkeynotthisstring'\n For that, use usethis::edit_r_environ().")
```
Comment (sckott) [inline R/key_helpers.R]: Okay if we remove the periods at the ends of `use usethis::edit_r_environ().` ? making it a little easier to copy/paste without including a period that will make it fail, same for others

**Comment by sckott** [comment]
hehe, forgot about https://github.com/ropensci/taxize/issues/682 - this can clean up that issue as well

**Comment by sckott** [comment]
For testing, you can test:

- what messages are returned with `expect_message()`
- what value is returned with `expect_equal` if you want to do exact match or `expect_match` if you want to do a regex match on the URL

i don't know of a way to test interactive use case, but can ignore that for now

**Comment by maelle** [comment]
Thanks, will do next week

File: R/getkey.R (hunk 1 above)
Comment (maelle) [reply R/getkey.R]: that part was added back in the docs of getkey

**Comment by sckott** [comment]
thanks @maelle !

**Pull request merged**